src/layouts/PLayoutGrid/PLayoutGrid.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="p-layout-grid">
+    <div class="p-layout-grid__header">
+      <slot name="header">
+        <div class="p-layout-grid__header-leading">
+          <slot name="header-leading" />
+        </div>
+        <div class="p-layout-grid__header-trailing">
+          <slot name="header-trailing" />
+        </div>
+      </slot>
+    </div>
+    <div class="p-layout-grid__toolbar">
+      <slot name="toolbar">
+        <div class="p-layout-grid__toolbar-leading">
+          <slot name="toolbar-leading" />
+        </div>
+        <div class="p-layout-grid__toolbar-trailing">
+          <slot name="toolbar-trailing" />
+        </div>
+      </slot>
+    </div>
+    <main class="p-layout-grid__main">
+      <div v-if="$slots.leftRail" class="p-layout-grid__left-rail">
+        <slot name="leftRail" />
+      </div>
+      <div class="p-layout-grid__content">
+        <slot />
+      </div>
+      <div v-if="$slots.rightRail" class="p-layout-grid__right-rail">
+        <slot name="rightRail" />
+      </div>
```

Comment:
If you follow the convention you set with the header and toolbar slots, this feels a little cleaner:
```suggestion
      <div v-if="$slots.mainLeading" class="p-layout-grid__main-leading">
        <slot name="mainLeading" />
      </div>
      <div class="p-layout-grid__content">
        <slot />
      </div>
      <div v-if="$slots.mainTrailing" class="p-layout-grid__main-trailing">
        <slot name="mainTrailing" />
      </div>
```


I'd also recommend using the `useSlots` composition Vue provides as it's generally recommended over the $ accessor.
```vue
<script lang="ts" setup>
import { useSlots } from 'vue'

const slots = useSlots()
</script>
```

And then updating your other slots from kebab -> pascal case just so they're internally consistent (this is one of those weird situations where kebab case is generally prefered but is gross when trying to access slots since you have to do `slots['kebab-case-slot']`